CMakeLists.txt

```diff
@@ -105,6 +105,11 @@ if(NOT CC_OPTIMIZE)
 endif()
 
 if (MSVC)
+  if (OCOS_ENABLE_STATIC_LIB) 
+    add_compile_options(/MT$<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:d>)
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:Debug>")
```

Comment:
You can set the CMAKE_MSVC_RUNTIME_LIBRARY variable from cmake command line.  So this change is unnecessary.

Reply:
OCOS_ENABLE_STATIC_LIB means we want to build this repo's code as a static library. However, it doesn't mean they need to static link to VC Runtime.

Reply:
@es0m-msft, According to the description, this PR is to enable static linking to VCRuntime, but OCOS_STATIC_LIBRARY flag was created for generating the static library of ort-extensions.lib instead of DLL. Can you try @snnn suggestion to see if it meets your requirements?

Reply:
OCOS_STATIC_LIBRARY is different from OCOS_ENABLE_STATIC_LIBRARY?

Reply:
@snnn   unfortunately, you can't if you want to respect generator expressions (I tried!)
Would you like a different OPTION for this usecase?

Reply:
> OCOS_STATIC_LIBRARY is different from protobuf_MSVC_STATIC_RUNTIME?

Sorry for typo, what I meant is protobuf_MSVC_STATIC_RUNTIME. We may create a flag like OCOS_MSVC_STATIC_RUNTIME for this usage if the flag is better.

I just tried in my local dev machine, it works with -DCMAKE_MSVC... in build.bat, except for the issue here: https://github.com/microsoft/onnxruntime-extensions/blob/1f0c76cefaa9359b9fc74ff4b9ed93214486205a/cmake/externals/sentencepieceproject.cmake#L22 which hard-code protobuf_MSVC_STATIC_RUNTIME=OFF, and it needs to be fixed.

Reply:
I searched the code, the only occurrence of "OCOS_ENABLE_STATIC_LIB" is:
```cmake
if(NOT OCOS_ENABLE_STATIC_LIB AND CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
```
So, it has nothing to do with VC Runtime. OCOS_ENABLE_STATIC_LIB  has effects only when CMAKE_SYSTEM_NAME equals to "Emscripten", which does not use  vc runtime at all.  So I don't think the variable is used to control which VC Runtime to use. And we should not give the variable a new meaning(which will surprise the person who created this cmake option).

Reply:
ONNX Runtime's [build.py](https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/build.py) has a build option: `--enable_msvc_static_runtime`. It works by setting `CMAKE_MSVC_RUNTIME_LIBRARY` cmake variable to  `MultiThreaded$<$<CONFIG:Debug>:Debug`.  If you build this repo with the main ONNX Runtime repo, you can use that build option. Otherwise you can do the same thing when you invoke cmake. Therefore I think there is no need to introduce a new cmake variable to control this. Every cmake project should respect the `CMAKE_MSVC_RUNTIME_LIBRARY` cmake variable's value.

Reply:
And, as `CMAKE_MSVC_RUNTIME_LIBRARY` means "the default value of MSVC_RUNTIME_LIBRARY". In cmake files, we should not set `CMAKE_MSVC_RUNTIME_LIBRARY` as it will change the `default` value instead of the `actual` value: `MSVC_RUNTIME_LIBRARY`.

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html#variable:CMAKE_MSVC_RUNTIME_LIBRARY
If you follow the docs, you're supposed to set CMAKE_MSVC_RUNTIME_LIBRARY not MSVC_RUNTIME_LIBRARY.

Reply:
regarding your use of OCOS_ENABLE_STATIC_LIB -- I'd suggest writing some documentation on how to use the flag. It is unclear as you can see by it's usage. It's related to WASM only apparently. 
Also, instead of having some command line collection parameters assembled as per build.py, having a clear option to enable the required functionality might be better.